core/os/os_darwin.odin

```diff
@@ -794,14 +794,14 @@ _readlink :: proc(path: string) -> (string, Errno) {
 }
 
 absolute_path_from_handle :: proc(fd: Handle) -> (string, Errno) {
-	buf : [256]byte
-	res  := _unix__fcntl(fd, F_GETPATH, &buf[0])
-	if	res != 0 {
-		return "", Errno(get_last_error())
+	buf: [DARWIN_MAXPATHLEN]byte
+	_, err := fcntl(int(fd), F_GETPATH, int(uintptr(&buf[0])))
```

Comment:
This could probably use the `or_return` treatment.

Reply:
That does not seem to work with Errno because it is an int: `'or_return' expects an "optional ok" like value, or an n-valued expression where the last value is either a boolean or can be compared against 'nil', got Errno`

Reply:
No, I meant after your patch of having it use the wrapper it reads like this?
```odin
absolute_path_from_handle :: proc(fd: Handle) -> (string, Errno) {
	buf: [DARWIN_MAXPATHLEN]byte
	_, err := fcntl(int(fd), F_GETPATH, int(uintptr(&buf[0])))
	if err != ERROR_NONE {
		return "", err
	}

	path := strings.clone_from_cstring(cstring(&buf[0]))
	return path, err
}
```
So `or_return`ing it would be:
```odin
absolute_path_from_handle :: proc(fd: Handle) -> (string, Errno) {
	buf: [DARWIN_MAXPATHLEN]byte
	_ = fcntl(int(fd), F_GETPATH, int(uintptr(&buf[0]))) or_return
	path := strings.clone_from_cstring(cstring(&buf[0]))
	return path, ERROR_NONE
}
```
Does that work?